hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/StakingValidator.java

```diff
@@ -132,7 +132,9 @@ private void validateStakedId(
         if (stakedIdKind.equals("STAKED_ACCOUNT_ID")) {
             validateTrue(accountStore.getAccountById(requireNonNull(stakedAccountIdInOp)) != null, INVALID_STAKING_ID);
         } else if (stakedIdKind.equals("STAKED_NODE_ID")) {
-            validateTrue(networkInfo.nodeInfo(requireNonNull(stakedNodeIdInOp)) != null, INVALID_STAKING_ID);
+            requireNonNull(stakedNodeIdInOp);
+            validateTrue(stakedNodeIdInOp >= -1L, INVALID_STAKING_ID);
```

Comment:
It is _not too late_ to give `-1` a **proper name**, which should have been done much earlier.  E.g., `public static final long RESET_STAKING_SENTINAL_ID = 1L;` and then provide `private boolean isValidStakedNodeId(final long stakedNodeId) { return stakedNodeId >= 0 }` and `private boolean isAcceptableStakedNodeId(final long stakedNodeId) { return isValidStakedNodeId(stakedNodeId) || stakedNodeId == RESET_STAKING_SENTINAL_ID; }`.  Or something similar.  So that the you don't have to rely on the comments to understand the code, when it's possible to make it so the code is understandable.  I'd also go for a `private final long RESET_STAKING_ACCOUNT_ID = 0L;` for the same reason.

(Names could be improved, but that's the idea.)

Reply:
Thanks, David, we will fix it in the future because I think it is used in multiple places. I fixed it because I was paged on testnet about it